htdocs/core/class/commonobject.class.php

```diff
@@ -3237,6 +3237,10 @@ public function fetchObjectLinked($sourceid = null, $sourcetype = '', $targetid
 		$targetid = (!empty($targetid) ? $targetid : $this->id);
 		$sourcetype = (!empty($sourcetype) ? $sourcetype : $this->element);
 		$targettype = (!empty($targettype) ? $targettype : $this->element);
+		if(!empty($this->module)){
+		    $sourcetype=$this->module.":$sourcetype";
```

Comment:
The other part of code (for exemple field elementype in actioncomm, or defintion of language files) are using another convention name that is
targettype.'"@".$this->module
instead of what you suggest
$this->module.":".targettype

Can you try to change your code to use this other syntax. I would like to have a common way of working for such needs. Also the "@module" is more intuitive since it means "at module"

Reply:
solved